api4/user.go

```diff
@@ -2381,3 +2384,101 @@ func verifyUserEmailWithoutToken(c *Context, w http.ResponseWriter, r *http.Requ
 
 	w.Write([]byte(user.ToJson()))
 }
+
+func migrateAuthToLDAP(c *Context, w http.ResponseWriter, r *http.Request) {
+	props := model.StringInterfaceFromJson(r.Body)
+	from, ok := props["from"].(string)
+	if !ok {
+		c.SetInvalidParam("from")
+		return
+	}
+	if len(from) == 0 || (from != "email" && from != "gitlab" && from != "saml") {
```

Comment:
"office365" and "google" are additional auth methods we should probably be supporting here as well. Unless there is a reason not to.

Reply:
Actually I followed the existing implementation in the legacy `mattermost` cli command. I can add them if they are applicable 👍